examples/06_02_join/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+#![cfg(test)]
+#![feature(async_await)]
+
+struct Book;
+struct Music;
+async fn get_book() -> Book { Book }
+async fn get_music() -> Music { Music }
+
+mod naiive {
+use super::*;
+// ANCHOR: naiive
+async fn get_book_and_music() -> (Book, Music) {
+    let book = get_book().await;
+    let music = get_music().await;
+    (book, music)
+}
+// ANCHOR_END: naiive
+}
+
+mod other_langs {
+use super::*;
+// ANCHOR: other_langs
+// WRONG -- don't do this
+async fn get_book_and_music() -> (Book, Music) {
+    let book_future = get_book();
+    let music_future = get_music();
+    (book_future.await, music_future.await)
+}
+// ANCHOR_END: other_langs
+}
+
+mod join {
+use super::*;
+// ANCHOR: join
+use futures::join;
+
+async fn get_book_and_music() -> (Book, Music) {
+    let book_fut = get_book();
+    let music_fut = get_music();
+    join!(book_fut, music_fut)
```

Comment:
Off-topic: Should we rewrite join!/try_join! with proc-macro to allow passing expressions directly here?

Reply:
That would be nice! I was actually just thinking the same thing.

Reply:
Filed https://github.com/rust-lang-nursery/futures-rs/pull/1783